include/oneapi/dpl/pstl/utils.h

```diff
@@ -312,7 +315,10 @@ class __transform_if_binary_functor
     mutable _BinaryPred _M_pred;
 
   public:
-    explicit __transform_if_binary_functor(_BinaryOper __op, _BinaryPred __pred) : _M_oper(__op), _M_pred(__pred) {}
+    explicit __transform_if_binary_functor(_BinaryOper __op, _BinaryPred __pred)
```

Comment:
I believe this semantic assume that in any case we have one copy of this param on caller side.
May be better to use universal reference ```&&``` and ```std::forward``` ?